src/main/java/tech/toparvion/analog/service/choice/ChoicesPropertiesChangesListener.java

```diff
@@ -0,0 +1,74 @@
+package tech.toparvion.analog.service.choice;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.cloud.endpoint.RefreshEndpoint;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.stereotype.Service;
+import tech.toparvion.analog.service.choice.ChoicesAutoReloadConfiguration.FileWatcherProvider;
+
+import javax.annotation.Nullable;
+import java.nio.file.*;
+
+import static java.nio.file.StandardWatchEventKinds.OVERFLOW;
+import static tech.toparvion.analog.service.choice.ChoicesAutoReloadConfiguration.CHOICES_AUTO_RELOAD_EXECUTOR;
+
+/**
+ * @author Polyudov
+ * @since v0.14
+ */
+@Service
+@EnableAsync
+@ConditionalOnChoicesAutoReloadEnabled
+class ChoicesPropertiesChangesListener {
+  private static final Logger log = LoggerFactory.getLogger(ChoicesPropertiesChangesListener.class);
+
+  private final RefreshEndpoint refreshEndpoint;
+  private final FileWatcherProvider fileWatcherProvider;
+
+  @Autowired
+  ChoicesPropertiesChangesListener(RefreshEndpoint refreshEndpoint,
+                                   @Nullable FileWatcherProvider fileWatcherProvider) {
+    this.refreshEndpoint = refreshEndpoint;
+    this.fileWatcherProvider = fileWatcherProvider;
+  }
+
+  @Async(CHOICES_AUTO_RELOAD_EXECUTOR)
+  @EventListener(ApplicationReadyEvent.class)
+  public void watchFile() {
+    if (fileWatcherProvider == null) {
+      return;
```

Comment:
A line of DEBUG log here would help us to investigate why the listener hasn't started. It would divide the search field into two importantly distinct parts: the application code and the framework.